### PR TITLE
Debian

### PR DESCRIPTION
**Pull request by fepitre**

### CODE REVIEW

File: debian/postinst [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+set -e
+
+case "${1}" in
+    triggered)
+        for trigger in ${2}; do
+            case "${trigger}" in
+
+                /etc/xdg/xfce4/xinitrc)
+                    debug "Setting XFCE Qubes App Menus..."
+                    sed -i 's|XDG_MENU_PREFIX="xfce-"|XDG_MENU_PREFIX="xfce-qubes-"|g' /etc/xdg/xfce4/xinitrc
```
Comment (marmarek) [inline debian/postinst]: Looking at that xinitrc script, it sets XDG_MENU_PREFIX only if it isn't set already. This gives a hope for setting it somewhere earlier without modifying a file from another package. Perhaps `/etc/X11/xinit/xinitrc.d`?

File: xfce4-settings-qubes.spec.in [hunk 2]
```diff
@@ -110,7 +111,9 @@ fi
 %{_sysconfdir}/xdg/xfce4/xfconf/xfce-perchannel-xml/xfce4-keyboard-shortcuts.xml.qubes
 %{_sysconfdir}/xdg/autostart/xfce4-xss-lock.desktop
 %{_sysconfdir}/xdg/autostart/qubes-update-xfce-config.desktop
+%{_sysconfdir}/xdg/menus/xfce-applications.menu.qubes
```
Comment (marmarek) [inline xfce4-settings-qubes.spec.in]: This is not the case anymore, no?

File: xfce4-settings-qubes.spec.in [hunk 3]
```diff
@@ -85,12 +83,14 @@ sed -e '/! Qubes options begin/,/! Qubes options end/c \
...
 update-xscreensaver-hacks
 
+
 %postun
 REPLACEFILE="${REPLACEFILE} %{_sysconfdir}/xdg/xfce4/panel/default.xml.qubes"
 REPLACEFILE="${REPLACEFILE} %{_sysconfdir}/xdg/xfce4/xfconf/xfce-perchannel-xml/xsettings.xml.qubes"
 REPLACEFILE="${REPLACEFILE} %{_sysconfdir}/xdg/xfce4/xfconf/xfce-perchannel-xml/xfce4-session.xml.qubes"
 REPLACEFILE="${REPLACEFILE} %{_sysconfdir}/xdg/xfce4/xfconf/xfce-perchannel-xml/xfce4-power-manager.xml.qubes"
 REPLACEFILE="${REPLACEFILE} %{_sysconfdir}/xdg/xfce4/xfconf/xfce-perchannel-xml/xfce4-keyboard-shortcuts.xml.qubes"
+REPLACEFILE="${REPLACEFILE} %{_sysconfdir}/xdg/menus/xfce-applications.menu.qubes"
```
Comment (marmarek) [inline xfce4-settings-qubes.spec.in]: Same as in %files - not here anymore.

**Review by marmarek — CHANGES_REQUESTED** [review]

File: xfce4-settings-qubes.spec.in (hunk 2 above)
Comment (fepitre) [reply xfce4-settings-qubes.spec.in]: Yes it's wrongly has not been included in the commit. Sorry.

File: xfce4-settings-qubes.spec.in (hunk 3 above)
Comment (fepitre) [reply xfce4-settings-qubes.spec.in]: Same here :)

File: debian/postinst (hunk 1 above)
Comment (fepitre) [reply debian/postinst]: Yes like we do in the case of XFCE template: https://github.com/fepitre/qubes-gui-agent-linux/blob/debian-guivm/appvm-scripts/etc/X11/Xsession.d/60xfce-desktop

**Comment by marmarek** [comment]
Now the produced Debian package is empty...

As for your earlier problem with dh_install complaining, this is actually pretty interesting corner case. By looking closely at the content of `debian/tmp` and `debian/xfce4-settings-qubes` directories, you can see that the `etc/X11/Xsession.d/55xfce-qubes` file is in both of them. The error message says it isn't installed, but the presence in the latter dir actually means it is installed. Another interesting thing is the file in `debian/tmp` has mode 0755, but the one in `debian/xfce4-settings-qubes` has 0664. This seems to be caused by a legacy behavior of `dh_install` - without explicit `--sourcedir` option it first looks for a file in the current directory, and only then in `debian/tmp`. And it happens that `etc/X11/Xsession.d/55xfce-qubes` is also available in the source root dir; and it has 0664 mode there. This also means `dh_install` copies the file from the current dir, not `debian/tmp`. And probably complains about not handled file in `debian/tmp` exactly because of this.
Ths solution seems to be as easy as `dh_install --sourcedir=debian/tmp --fail-missing`.

**Comment by fepitre** [comment]
> dh_install --sourcedir=debian/tmp --fail-missing

Yes indeed this corner case is very weird and interesting. Thank you for the investigation. I'll readd the install file and fix the rules.

**Comment by fepitre** [comment]
Looking to other Xsession.d files they are 644 but that still makes the problem persisting if `--sourcedir=debian/tmp` is not specified. Now it should be good.

File: xfce4-settings-qubes.spec.in [hunk 4]
```diff
@@ -1,73 +1,66 @@
...
 
 Requires:	qubes-artwork
 Requires:	xfce4-panel
 Requires:	xss-lock
 Requires(post):	xfce4-panel
 
+# We ship garcon menu file into this package
+# the fork is not needed anymore
+Conflicts:  garcon = 1000:0.6.4
+Requires: garcon >= 0.6.2
```
Comment (marmarek) [inline xfce4-settings-qubes.spec.in:20]: This doesn't work for me. On update (`qubes-dom0-update --clean`) I get:
```
 Problem: package xfce4-settings-qubes-4.0.3-2.1.fc32.x86_64 requires garcon >= 0.6.2, but none of the providers can be installed
  - cannot install the best update candidate for package garcon-1000:0.6.4-4.fc32.x86_64
  - package xfce4-settings-qubes-4.0.3-2.1.fc32.x86_64 conflicts with garcon = 1000:0.6.4 provided by garcon-1000:0.6.4-4.fc32.x86_64
  - problem with installed package xfce4-settings-qubes-4.0.3-2.fc32.x86_64
  - cannot install the best update candidate for package xfce4-settings-qubes-4.0.3-2.fc32.x86_64
```

I've tried also with `--allowerasing`, `--best`, `--nobest`, `--action=distro-sync` and nothing helped. BTW `--best --allowerasing` is [hardcoded on the VM side](https://github.com/QubesOS/qubes-core-agent-linux/blob/master/package-managers/qubes-download-dom0-updates.sh#L62), which makes `--nobest` refused.

**Review by marmarek — CHANGES_REQUESTED** [review]

**Comment by marmarek** [comment]
We haven't found elegant solution for the clean upgrade. The hacky one is running `rpm -e --nodeps garcon` before the upgrade.

**Pull request merged**